File: java-codellama7B-AWQ-all/taskid-161_solve-gen18.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    // You are given a string s.
    // if s[i] is a letter, reverse its case from lower to upper or vise versa, 
    // otherwise keep it as it is.
    // If the string contains no letters, reverse the string.
    // The function should return the resulted string.
    // Examples
    // >>> solve(("1234"))
    // ("4321")
    // >>> solve(("ab"))
    // ("AB")
    // >>> solve(("#a@C"))
    // ("#A@c")
    public static String solve(String s) {

        int i = 0;
        String res = "";
        while (i < s.length()) {
            if (s.charAt(i) >= 65 && s.charAt(i) <= 90) {
                res += (char) (s.charAt(i) + 32);
            } else if (s.charAt(i) >= 97 && s.charAt(i) <= 122) {
                res += (char) (s.charAt(i) - 32);
            } else {
                res += s.charAt(i);
            }
            i++;
        }
        if (res.isEmpty()) {
            res = s;
        }
        return res;
    }
    public static void main(String[] args) {
    assert(solve(("AsDf")).equals(("aSdF")));
    assert(solve(("1234")).equals(("4321")));
    assert(solve(("ab")).equals(("AB")));
    assert(solve(("#a@C")).equals(("#A@c")));
    assert(solve(("#AsdfW^45")).equals(("#aSDFw^45")));
    assert(solve(("#6@2")).equals(("2@6#")));
    assert(solve(("#$a^D")).equals(("#$A^d")));
    assert(solve(("#ccc")).equals(("#CCC")));
    }

}

File: py-codellama34B-FP-all/taskid-161_solve-gen18.py
def solve(s: str) -> str:
    """You are given a string s.
    if s[i] is a letter, reverse its case from lower to upper or vise versa, 
    otherwise keep it as it is.
    If the string contains no letters, reverse the string.
    The function should return the resulted string.
    Examples
    >>> solve('1234')
    '4321'
    >>> solve('ab')
    'AB'
    >>> solve('#a@C')
    '#A@c'
    """

    result = ""
    for char in s:
        if char.isalpha():
            result += char.lower() if char.isupper() else char.upper()
        else:
            result += char
    return result[::-1] if result == "" else result
